scripts/run_inference.py

```diff
@@ -183,13 +183,30 @@ def load(
                 try:
                     if base_model_name_or_path is None:
                         raise ValueError("base_model_name_or_path has to be passed")
-                    base_model = AutoModelForCausalLM.from_pretrained(
-                        base_model_name_or_path,
-                        attn_implementation="flash_attention_2"
-                        if use_flash_attn
-                        else None,
-                        torch_dtype=torch.bfloat16 if use_flash_attn else None,
+
+                    is_quantized = os.path.exists(
+                        os.path.join(base_model_name_or_path, "quantize_config.json")
                     )
+                    if is_quantized:
+                        gptq_config = GPTQConfig(bits=4, exllama_config={"version": 2})
+                        base_model = AutoModelForCausalLM.from_pretrained(
+                            base_model_name_or_path,
+                            attn_implementation="flash_attention_2"
+                            if use_flash_attn
+                            else None,
+                            device_map="cuda:0",
+                            torch_dtype=torch.float16 if use_flash_attn else None,
```

Comment:
Does this always have to be torch.float16 and is this relevant to flash-attn being set? I know Fabian and Aaron set float16, is this something needed for loading quantized models?

I know in the fms-acceleration docs:
> When setting --auto_gptq triton_v2 plus note to also pass --torch_dtype float16 and --fp16, or an exception will be raised. This is because these kernels only support this dtype.

Is this also true here for loading the model?

Reply:
@anhuong we are in the process of making a new plugin release, and we re-opened an investigation on the exact conditions this will be needed. We may have the flexibility to also run `auto_gptq triton_v2` in bfloat16. We will report more when we find out very soon. 

But in the meantime, we have tested `float16` extensitively and can confirm the above setting will work, but it will be nice if this restriction can be also relaxed.

Reply:
Since we don't use fms-acceleration, for loading tuned models for evaluation, then using the kernels available from HF/AutoGPTQ directly will require dtype to be set to float16

Reply:
@anhuong sorry i think our responses are abit confusing

1. in the fms-accel docs, we set `--auto_gptq triton_v2` because we are using different kernels for training (i.e., not `exllama` as in `run_inference.py`. 
2. when usin`--auto_gptq triton_v2` we specified in docs that use `float16`, but this is under investigation and could be relaxed.
3. when using `exllama`, you must use `float16`. However, the trained checkpoint if loaded in some other inference framework (.e.g, vllm) can be loaded in other types as they are supported. 

To conclude, the type depends on what inference framework you load the checkpoint in. Here `run_inference` loads in `exllama`, so you must respect the type it supports.

Reply:
thank you, these details are very helpful to understand

Reply:
Based on these details above it sounds like this should always be set to float16 even if flash-attn is not used, @aluu317

Reply:
Ahh yup, updated!